### ISSUE TITLE
Support for multiple models

### ISSUE THREAD

**Issue by sublunar84** [dotnet/machinelearning#6667]
I try to add two different models with AddPredictionEnginePool like below:
`s.AddPredictionEnginePool<MLxxxData, MLxxxPrediction>()
	.FromFile(modelName: "mlNetxxxModel", filePath: GetModelFilePath(CheckType.xxx), watchForChanges: false);

s.AddPredictionEnginePool<MLyyyData, MLyyyPrediction>()
	.FromFile(modelName: "mlNetyyyModel", filePath: GetModelFilePath(CheckType.yyy), watchForChanges: false);`

Only the last model is added though. If I try to use the first one I get an error. Shouldn't you be able to load more than one model? Instead of having to make one application for each model. Cannot find any information on this problem.

**Comment by luisquintanilla** [comment]
Hi @sublunar84

Sorry to hear you ran into issues. Would you be able to show what the error you're getting is. You're right that you should be able to have multiple models in a single app, but we'd need more information to investigate. 

Thanks!

**Comment by sublunar84** [comment]
Hi, 
Yes, I can explain a bit more.

I have an Azure function app in .NET 6. In Program.cs two calls to AddPredictionEnginePool is made for two different models. Like so:

```
public class Program
	{
		public static void Main()
		{
			var host = new HostBuilder()
				.ConfigureFunctionsWorkerDefaults()
				.ConfigureServices(s =>
				{
					s.AddPredictionEnginePool<MLSecuritySheetData, MLSecuritySheetPrediction>()
						.FromFile(modelName: "mlNetSDBModel", filePath: GetModelFilePath(CheckType.SecuritySheet), watchForChanges: false);

                                        s.AddPredictionEnginePool<MLEpdData, MLEpdPrediction>()
						.FromFile(modelName: "mlNetEPDModel", filePath: GetModelFilePath(CheckType.Epd), watchForChanges: false);

					s.AddSingleton<SafetySheetProcessor>();
					s.AddSingleton<EpdProcessor>();
					s.AddSingleton<Settings>();
				})
				.Build();

			host.Run();
		}
```

The Azure function has a Queue-trigger. In the queue there are items of two different types: SDB and EPD checks. Depending on what type is in the message, one of the two models is being used to make predicitions.

The error occurs in the call to "predictionEnginePool.Predict" method.

In the above example (where the mlNetEPDModel is defined after the mlNetSDBModel in ConfigureServices), the error occurs when trying to use mlNetSDBModel to make a predicition:

![image](https://github.com/dotnet/machinelearning/assets/50997233/db93fdc0-55a6-40a4-a1de-a7b951daa225)

The error says "Could not find input column 'Är EPD'". This column can be found in class MLEpdData - which belongs to model mlNetEPDModel. If I change order in ConfigureServices, I get a similar error when trying to use the mlNetEPDModel. But then the mlNetSDBModel can be used ok to make predicitions.

So, my conclusion is that with this setup I can only use one of the models - the one that is defined last in ConfigureServices. How should I configure it differently so I can use both models from my Azure function?

Kind regards
Helena

**Comment by sublunar84** [comment]
Hi, any updates on this?

**Comment by sublunar84** [comment]
Hi again,
I finally figured out what the problem was so I will share it in case anyone else has a similar problem.

Basically, I use FromFile to add a model to the AddPredictionEnginePool-function. The filename is calculated in my method GetModelFilePath. In this method I read the model file in question (the zip-file) from BlobStorage. Two different models from two different blobs in blobStorage. But then I saved the models in Azure function temporary storage with the same file name. As soon as I changed this to use two different file names for each model it worked as expected.
